uperf_wrapper/trigger_uperf.py

```diff
@@ -0,0 +1,192 @@
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import re
+import subprocess
+from datetime import datetime
+
+import numpy as np
+
+
+class Trigger_uperf():
+    def __init__(self, args):
+        self.uuid = args.uuid
+        self.user = args.user
+        self.clientips = args.clientips
+        self.remoteip = args.remoteip
+        self.hostnetwork = args.hostnetwork
+        self.serviceip = args.serviceip
+        self.server_node = args.server_node
+        self.client_node = args.client_node
+        self.cluster_name = args.cluster_name
+        self.workload = args.workload
+        self.run = args.run
+        self.resourcetype = args.resourcetype
+
+    def _json_payload(self, data, iteration, uuid, user, hostnetwork, serviceip, remote, client,
+                      clustername, resource_type, server_node, client_node):
+        processed = []
+        prev_bytes = 0
+        prev_ops = 0
+        prev_timestamp = 0.0
+        for result in data['results']:
+            norm_ops = int(result[2]) - prev_ops
+            if norm_ops == 0:
+                norm_ltcy = 0.0
+            else:
+                norm_ltcy = ((float(result[0]) - prev_timestamp) / (norm_ops)) * 1000
+            processed.append({
+                "workload": "uperf",
+                "uuid": uuid,
+                "user": user,
+                "cluster_name": clustername,
+                "hostnetwork": hostnetwork,
+                "iteration": int(iteration),
+                "remote_ip": remote,
+                "client_ips": client,
+                "uperf_ts": datetime.fromtimestamp(int(result[0].split('.')[0]) / 1000),
+                "test_type": data['test'],
+                "protocol": data['protocol'],
+                "service_ip": serviceip,
+                "message_size": int(data['message_size']),
+                "num_threads": int(data['num_threads']),
+                "duration": len(data['results']),
+                "bytes": int(result[1]),
+                "norm_byte": int(result[1]) - prev_bytes,
+                "ops": int(result[2]),
+                "norm_ops": norm_ops,
+                "norm_ltcy": norm_ltcy,
+                "kind": str(resource_type),
+                "client_node": client_node,
+                "server_node": server_node
+            })
+            prev_timestamp = float(result[0])
+            prev_bytes = int(result[1])
+            prev_ops = int(result[2])
+        return processed
+
+    def _run_uperf(self, workload):
+        cmd = "uperf -v -a -x -i 1 -m {}".format(workload)
+        process = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+        stdout, stderr = process.communicate()
+        return stdout.strip().decode("utf-8"), process.returncode
+
+    def _parse_stdout(self, stdout):
+        # This will effectivly give us:
+        # ripsaw-test-stream-udp-16384
```

Comment:
Small nit against myself here... but we don't have the nthr in the output here, but in the code we now have nthr :D